tests/test_templates.py

```diff
@@ -123,7 +125,17 @@
                               (QAOAEmbedding,
                                {'features': [1., 2.],
                                 'weights': [[0.1, 0.1, 0.1]]},
-                               {})
+                               {}),
+                              (broadcast,
+                               {'parameters': [[1.], [1.]]},
+                               {'block': qml.RX,
+                                'wires': [0, 1],
+                                'structure': 'single'}),
+                              (broadcast,
+                               {'parameters': [[1., 1., 1.]]},
+                               {'block': qml.CRot,
+                                'wires': [0, 1],
+                                'structure': 'double'})
```

Comment:
It might be worth adding a case where:
* `'block'` is user defined template here
* `'structure'` is `double-odd`

Reply:
I was thinking about that and actually tend towards only testing ``broadcast`` with one structure here. These are just integration tests, and the structures themselves working as expected should be fully tested in ``test_templates_broadcast``.

Also, there is a quadratic growth in the number of tests run, since every template gets integrated with every other.

There is a subsequent PR adding more structures where I removed the `double` test case, so if you agree to the above we can leave it as is for now and change it in the next PR.

Reply:
Sounds good! :)